ReSharper.FSharp/src/FSharp.Psi.Features/src/Intentions/ToInterpolatedString.fs

```diff
@@ -0,0 +1,155 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Intentions
+
+open System.Text
+open JetBrains.DocumentModel
+open JetBrains.ReSharper.Feature.Services.ContextActions
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Util
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Feature.Services.Daemon
+
+type private StringManipulation =
+    | InsertInterpolation of specifier: string * expr: IFSharpExpression
+    | EscapeBrace of braceChar: char
+
+[<ContextAction(Name = "ToInterpolatedString", Group = "F#",
+                Description = "Convert an printf-style format string with an interpolated string")>]
+type ToInterpolatedStringAction(dataProvider: FSharpContextActionDataProvider) =
+    inherit FSharpContextActionBase(dataProvider)
+
+    let [<Literal>] opName = "ToInterpolatedStringAction"
+
+    override x.Text = "To interpolated string"
+
+    override x.IsAvailable _ =
+        let prefixAppExpr = dataProvider.GetSelectedElement<IPrefixAppExpr>()
+        if isNull prefixAppExpr then false else
+
+        // todo: fix NRE in tests
+        //if not (FSharpLanguageLevel.isFSharp50Supported prefixAppExpr) then false else
+
+        let literalExpr = prefixAppExpr.ArgumentExpression.IgnoreInnerParens().As<ILiteralExpr>()
+        if isNull literalExpr then false else
+
+        // todo: support FSharpTokenType.TRIPLE_QUOTED_STRING and FSharpTokenType.VERBATIM_STRING
+        let tokenType = literalExpr.Literal.GetTokenType()
```

Comment:
You should always assume that generated child node properties like `Literal` may return `null` due to error in file. There's a `getTokenType` function that does the same while is safe to use here.